machine-learning/src/main.py

```diff
@@ -38,11 +41,13 @@ class ClipRequestBody(BaseModel):
 
 @app.on_event("startup")
 async def startup_event():
+    if not eager_startup:
```

Comment:
I would like to keep the eager start-up as default. From our testing, having the model load during runtime can run into multiple instances trying to download and load the model into memory, therefore inflating the RAM usage of the container

Reply:
Alternatively, would it be possible to create some sort of cross process lock that would allow the models to be loaded lazily, but only downloaded and subsequently loaded into once? Basically best of both worlds.

Reply:
`eager_startup` is enabled by default so it should still work the same in terms of loading all models at startup. 

I agree that downloading at runtime but loading lazily would be even better. The simplest way to do that would be to just call `_get_model` and `del` the loaded model right after. (In the CUDA PR, this could be done with `_prepare_hf_model` to compile without loading them).

A more elegant way to go about it would be to use something like [huggingface_hub](https://huggingface.co/docs/huggingface_hub/main/en/guides/download) and whatever `insightface` uses, checking if the files are there and downloading if not. It would be a bit more work to implement, though.

Reply:
If this works the same as it does now that's fine and we can look into those options in another PR.

Reply:
I tweaked it so it still loads models if eager startup is disabled, but they get garbage collected right after since they don't get added to the model cache. Eager startup works the same as usual.